src/linear_algebra/affine_space/finite_dimensional.lean

```diff
@@ -305,22 +305,41 @@ end
 at most `1`. -/
 def collinear (s : set P) : Prop := module.rank k (vector_span k s) ≤ 1
 
+variables {k}
+
 /-- The definition of `collinear`. -/
-lemma collinear_iff_dim_le_one (s : set P) : collinear k s ↔ module.rank k (vector_span k s) ≤ 1 :=
+lemma collinear_iff_dim_le_one {s : set P} : collinear k s ↔ module.rank k (vector_span k s) ≤ 1 :=
 iff.rfl
 
 /-- A set of points, whose `vector_span` is finite-dimensional, is
 collinear if and only if their `vector_span` has dimension at most
 `1`. -/
-lemma collinear_iff_finrank_le_one (s : set P) [finite_dimensional k (vector_span k s)] :
+lemma collinear_iff_finrank_le_one {s : set P} [finite_dimensional k (vector_span k s)] :
   collinear k s ↔ finrank k (vector_span k s) ≤ 1 :=
 begin
-  have h := collinear_iff_dim_le_one k s,
+  have h : collinear k s ↔ module.rank k (vector_span k s) ≤ 1 := collinear_iff_dim_le_one,
```

Comment:
This seems like a regression to me. Isn't it better to keep the arguments explicit?
(Of course, writing out the type explicitly is fine. But being *forced* to do that is less ideal.)

Reply:
I'm not actually entirely sure when such lemmas repeating a definition should be present at all; there may be a case for not having the lemma, and just using `collinear` in rewrites and `iff.rfl` (with explicit type) in cases such as this where the actual statement is needed. But if there is some exception to the general rule about making arguments implicit when they appear on both sides of an `iff` lemma (which is what I was trying to follow here for `collinear` lemmas in general), then indeed the arguments could be kept explicit.

Reply:
I think that in tactic mode, using `rw`, it almost never hurts to have explicit variables on an iff lemma.
But in term mode, when writing `(some_lemma _ _ _ _ _).mpr, all those underscores quickly become old.
And it's exactly when you want to `have` some iff statement out of the blue where these explicit variables really shine.
So it's a bit of a balancing act. I'm not sure if I can write a precise rule.

Reply:
I've reverted the change to argument implicitness for `collinear_iff_dim_le_one`. (Even the general rules for when arguments are explicit or implicit don't actually seem to be documented in the mathlib style guidelines.)

As noted on Zulip, this PR shouldn't be merged in the same batch as #16191, although either of these two PRs could be merged first, because the change to argument implicitness for `collinear_iff_of_mem` will require one use of that lemma in that PR to be updated.